android/FreshchatPlugin.java

```diff
@@ -0,0 +1,203 @@
+
+  private Context appContext;
+  private Activity appActivity;
+  private FreshchatUser fchatUser;
+
+  private final String TAG = "{freshchat}";
+
+  public void onCreateApplication(Context applicationContext) {
+    appContext = applicationContext;
+  }
+
+  public void onCreate(Activity activity, Bundle savedInstanceState) {
+    String appId = "";
+    String appKey = "";
+
+    PackageManager manager = activity.getBaseContext().getPackageManager();
+
+    appActivity = activity;
+
+    logger.log(TAG + "inside onCReate");
+    try {
+      Bundle meta = manager.getApplicationInfo(activity.getApplicationContext().getPackageName(),
+        PackageManager.GET_META_DATA).metaData;
+      appId = meta.get("FRESHCHAT_APP_ID").toString();
+      appKey = meta.get("FRESHCHAT_APP_KEY").toString();
+    } catch (Exception ex) {
+      logger.log(TAG + "{exception}", "" + ex.getMessage());
+    }
+
+    logger.log(TAG + "before config init app: " + appId + " app key: " + appKey);
+    FreshchatConfig fchatConfig = new FreshchatConfig(appId, appKey);
+    // TODO: Read these from config
+    fchatConfig.setGallerySelectionEnabled(false);
+    fchatConfig.setCameraCaptureEnabled(false);
+
+    Freshchat.getInstance(appContext).init(fchatConfig);
+    fchatUser = Freshchat.getInstance(appContext).getUser();
+    logger.log(TAG + "end of onCreate");
+  }
+
+  public void onResume() {
+  }
+  
+  public void onRenderResume() {
+  }
+  
+  public void onStart() {
+  }
+  
+  public void onFirstRun() {
+  }
+  
+  public void onPause() {
+  }
+  
+  public void onRenderPause() {
+  }
+  
+  public void onStop() {
+  }
+  
+  public void onDestroy() {
+  }
+  
+  public void onNewIntent(Intent intent) {
+  }
+  
+  public void setInstallReferrer(String referrer) {
+  }
+  
+  public void onActivityResult(Integer request, Integer result, Intent data) {
+  }
+  
+  public boolean consumeOnBackPressed() {
+    return true;
+  }
+  
+  public void onBackPressed() {
+  }
+
+  public void setName (String param) {
+    JSONObject reqJson;
+
+    try {
+      reqJson = new JSONObject(param);
+      fchatUser.setFirstName(reqJson.getString("first_name"));
+      fchatUser.setLastName(reqJson.getString("last_name"));
+    } catch (Exception e) {
+      logger.log(TAG + "{exception}", "" + e.getMessage());
+    }
+  }
+
+  public void setEmail (String param) {
+    JSONObject reqJson;
+
+    try {
+      reqJson = new JSONObject(param);
+      fchatUser.setEmail(reqJson.getString("email"));
+    } catch (Exception e){
+      logger.log(TAG + "{exception}", "" + e.getMessage());
+    }
+  }
+
+  public void setExternalId (String param) {
+    JSONObject reqJson;
+
+    try {
+      reqJson = new JSONObject(param);
+      Freshchat.getInstance(appContext).identifyUser(reqJson.getString("id"),
+        null);
+    } catch (Exception e){
+      logger.log(TAG + "{exception}", "" + e.getMessage());
+    }
+  }
+
+  public void addMetaData (String param) {
```

Comment:
is this for events ?

Reply:
nope. its for adding custom user properties @rampr